src/xrpld/app/misc/WasmVM.h

```diff
@@ -122,12 +149,217 @@ struct HostFunctions
     virtual ~HostFunctions() = default;
 };
 
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+enum WasmTypes { WT_I32, WT_I64, WT_F32, WT_F64 };
+
+struct WasmImportFunc
+{
+    std::string name;
+    std::optional<WasmTypes> result;
+    std::vector<WasmTypes> params;
+    void* udata = nullptr;
+    // wasm_func_callback_with_env_t
+    void* wrap = nullptr;
+};
+
+#define WASM_IMPORT_FUNC(v, f, ...) \
+    WasmImpFunc<f##_proto>(         \
+        v, #f, reinterpret_cast<void*>(&f##_wrap), ##__VA_ARGS__);
```

Comment:
This macro is hard to read - if it's critical to the code, can you provide an example above of what it might look like?

Reply:
usage in runEscrowWasm function